src/core/lib/surface/call.c

```diff
@@ -427,6 +429,8 @@ static void destroy_call(grpc_exec_ctx *exec_ctx, void *call,
 
   get_final_status(call, set_status_value_directly,
                    &c->final_info.final_status);
+  c->final_info.stats.latency =
+      gpr_time_sub(gpr_now(GPR_CLOCK_MONOTONIC), c->start_time);
 
```

Comment:
I like asserting. In part for documentation purposes, in part because I have a (n involuntary) thing for introducing bugs. Just a soft suggestion to add `GPR_ASSERT( gpr_time_cmp(c->final_info.stats.latency, gpr_time_0(GPR_CLOCK_MONOTONIC) > 0)` after the new code.